### ISSUE TITLE
Chunking UploadFile: read(byte_count) awaits until upload is fully complete

### ISSUE THREAD

**Issue by dorinclisu** [tiangolo/fastapi#3136]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
import logging

from fastapi import FastAPI
from fastapi import File, UploadFile


app = FastAPI()

@app.post('/video')
async def upload_video(
    video: UploadFile = File(...)
):
    chunk = await video.read(100)
    logging.warning('Read the first 100 bytes')

    remaining = await video.read()
    logging.warning('Read all the file')
```

### Environment
* OS: Linux
* FastAPI Version: 0.63.0

### Description
Just run the example and try uploading a large file (100MB or above) from Swagger UI.
The expected behavior is that the first bytes are read quickly, before the upload is complete.

Unfortunately, the behavior I see is that it takes a rather long time for the whole file to upload, and only then the first chunked read is available, immediately followed by the second read.  This kind of defeats the whole purpose of chunked read.

Is it not possible to start reading the file before upload is complete? The reason I'm asking is that I'd like to start uploading to s3 in parallel, to shorten the response wait time.

**Comment by Dustyposa** [comment]
You can solve it in [here](https://github.com/tiangolo/fastapi/issues/58)

**Comment by dorinclisu** [comment]
@Dustyposa I am not sure what exactly you refer to. The linked issue mentions 3 aspects:
- `UploadFile` - this is what I am already doing and not working well
- `StreamingResponse` - out of scope
- Using `Request` directly - https://fastapi.tiangolo.com/advanced/using-request-directly/

Assuming you mean the 3rd, the example is:
```Python
async for chunk in request.stream():
    ...
```

Does this mean that I need to do the form decoding myself, figure out where the file data starts and then handle the rest?

**Comment by Dustyposa** [comment]
Yes, u should use `stream` for your  demand.

**Comment by hall-b** [comment]
@dorinclisu did you manage to do the form decoding yourself? If yes, could you share your code snippet ?
